api/clients/codecs/blob.go

```diff
@@ -0,0 +1,79 @@
+package codecs
+
+import (
+	"fmt"
+)
+
+// Blob is data that is dispersed on eigenDA.
+//
+// A Blob is represented under the hood by a coeff polynomial
+type Blob struct {
+	coeffPolynomial *coeffPoly
+	// blobLength must be a power of 2, and should match the blobLength claimed in the BlobCommitment
+	// This is the blob length IN SYMBOLS, not in bytes
+	blobLength uint32
+}
+
+// BlobFromBytes initializes a Blob from bytes
+//
+// blobLength is the length of the blob IN SYMBOLS
+func BlobFromBytes(bytes []byte, blobLength uint32) (*Blob, error) {
+	poly, err := coeffPolyFromBytes(bytes)
+	if err != nil {
+		return nil, fmt.Errorf("polynomial from bytes: %w", err)
+	}
+
+	return BlobFromPolynomial(poly, blobLength)
+}
+
+// BlobFromPolynomial initializes a blob from a polynomial
+//
+// blobLength is the length of the blob IN SYMBOLS
+func BlobFromPolynomial(coeffPolynomial *coeffPoly, blobLength uint32) (*Blob, error) {
+	return &Blob{
+		coeffPolynomial: coeffPolynomial,
+		blobLength:      blobLength,
+	}, nil
+}
+
+// GetBytes gets the raw bytes of the Blob
+func (b *Blob) GetBytes() []byte {
+	return b.coeffPolynomial.getBytes()
+}
```

Comment:
have a preference for `Bytes()` without the `get` prefix

Reply:
or `toBytes()` to be consistent with `ToPayload`?

Reply:
I personally tend to understand a nuanced difference between `To` and `Get`

-`to` (or in other contexts, `compute`) seems to indicate that the data is changing form, or being processed in some way
- `get` implies that the method is simply fetching something that already exists, byte for byte

So, I chose `ToPayload` because the method is transforming the data, to produce a payload. And it's `GetBytes`, because we are returning the literal bytes, as they already exist.

If this nuance isn't something that's generally understood, though, I'm ok changing the current names. What would your preferred naming be for these two methods? `Bytes()` and `ToPayload()`?

Reply:
@samlaf How do you feel about using the `serialize` / `deserialize` nomenclature? https://github.com/Layr-Labs/eigenda/pull/1120#discussion_r1954921604

Reply:
Oh yes that’s even better!

Reply:
[f1a4e85f](https://github.com/Layr-Labs/eigenda/pull/1120/commits/f1a4e85f83ef84935aaba888cbb28dfb9e743612)

Reply:
Just realized now that golang has standard interfaces for serialization: MarshalJSON() and [MarshalBinary()](https://pkg.go.dev/encoding#BinaryMarshaler). Might be worth it to add a comment as to why we define our own function that isn't marshalBinary (perhaps we should just implement MarshalBinary() instead? so that people could encode our structs to gob if they need to for whatever reason).

Also the fact that we need that extra length param to unmarshal is another consequence of this weird "edge case that'll never happen". If we changed that requirement then this code would also be much simpler and more standard go. Again, we can discuss (perhaps with your brother).. I might be missing something.

Reply:
I don't think it's worth implementing an interface just because it exists, without a concrete use case. I suppose it would be fine to add a comment saying why we don't implement the interface, but it doesn't seem necessary: throughout our whole codebase, there are no implementations of `BinaryMarshaler`: I think the implicit explanation for why it's not implemented is that we don't need it.

Reply:
I don't feel strongly one way or another about the golang interfaces, but I'm generally in favor of avoiding code complexity that doesn't have a justification on the short-to-medium term roadmap.

Perhaps also relevant, the string `MarshalBinary` doesn't appear anywhere in the eigenDA repo right now, so this isn't a pattern we are currently using.

Reply:
SGTM. Let's resolve this bike shedding haha. I just think that we should abide by golang conventions because otherwise it makes the code harder to read for people familiar with those conventions. The convention here being the use of `Marshal` instead of `Serialize`.